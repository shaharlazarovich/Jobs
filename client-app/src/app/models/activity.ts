export interface IAttendee {
    userName: string;
    displayName: string;
    image: string;
    isHost: boolean;
    following?: boolean,
}

//we are adding this interface to mimic what we do on the server for paging -
//which is to have an entity around our activity array, in order to add a count
//of activities - which then, we could device in the number of pages
export interface IActivitiesEnvelope {
    activities: IActivity[];
    activityCount: number
}

export interface IActivity {
    id: string;
    title: string;
    description: string;
    category: string;
    date: Date;
    city: string;
    venue: string;
    isGoing: boolean;
    isHost: boolean;
    attendees: IAttendee[];
    comments: IComment[];
}

export interface IComment {
    id: string;
    createdAt: Date;
    body: string;
    username: string;
    displayName: string;
    image: string;
}

export interface IActivityFormValues extends Partial<IActivity> {
    time?: Date    
}

export class ActivityFormValues implements IActivityFormValues {
    id?: string = undefined;
    title: string = '';
    category: string = '';
    description: string = '';
    date?:Date = undefined;
    time?: Date = undefined;
    city: string = '';
    venue: string = '';

    constructor(init?: IActivityFormValues) {
        if (init && init.date) {
            init.time = init.date;
        }
        Object.assign(this, init); //this will initialize our object of class ActivityFormValues
                                  // with the above values
    }
}