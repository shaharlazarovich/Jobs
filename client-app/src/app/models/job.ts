//we are adding this interface to mimic what we do on the server for paging -
//which is to have an entity around our jobs array, in order to add a count
//of jobs - which then, we could device in the number of pages
export interface IJobsEnvelope {
    jobs: IJob[];
    jobsCount: number
}

export interface IJob {
    id: number;
    jobName: string;
    company: string;
    replication: string;
    servers: string;
    lastRun: Date;
    date: Date;
    time: Date;
    rta: string;
    results: string;
    key: string;
    rtoNeeded: string;
    jobIP: string;
}


export interface IJobFormValues extends Partial<IJob> {
     
}

export class JobFormValues implements IJobFormValues {
    id?: number = 0;
    jobName: string = '';
    company: string = '';
    replication: string = '';
    servers: string = '';
    lastRun?:Date = undefined;
    date?:Date = undefined;
    time?:Date = undefined;
    rta?: string = '';
    results: string = '';
    key: string = '';
    rtoNeeded: string = '';
    jobIP: string = '';

    constructor(init?: IJobFormValues) {
        Object.assign(this, init); //this will initialize our object of class JobFormValues
                                  // with the above values
    }
}