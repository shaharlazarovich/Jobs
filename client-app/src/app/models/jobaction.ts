//we are adding this interface to mimic what we do on the server for paging -
//which is to have an entity around our jobs array, in order to add a count
//of jobs - which then, we could device in the number of pages
export interface IJobActionsEnvelope {
    jobActions: IJobAction[];
    jobActionsCount: number
}

export interface IJobAction {
    id: string;
    jobId: string;
    jobName: string;
    userId: string;
    actionDate: Date;
    remoteIP: string;
    remoteResponse: string;
    requestProperties: string;
    source: string;
    action: string;
}

export class JobAction implements IJobAction {
    id: string = '';
    jobId: string = '';
    jobName: string = '';
    userId: string = '';
    actionDate: Date = new Date();
    remoteIP: string = '';
    remoteResponse: string = '';
    requestProperties: string = '';
    source: string = '';
    action: string = '';

    constructor(init?: IJobAction) {
        Object.assign(this, init); //this will initialize our object of class JobActions
                                  // with the above values
    }
}