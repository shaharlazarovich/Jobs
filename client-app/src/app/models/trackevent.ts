//we are adding this interface to mimic what we do on the server for paging -
//which is to have an entity around our jobs array, in order to add a count
//of jobs - which then, we could device in the number of pages
export interface ITrackEventsEnvelope {
    trackEvents: ITrackEvent[];
    trackEventsCount: number
}

export interface ITrackEvent {
    id: string;
    jobId?: string;
    jobName?: string;
    userId?: string;
    actionDate: Date;
    remoteIP?: string;
    remoteResponse?: string;
    requestProperties?: string;
    source: string;
    event: string;
}

export class TrackEvent implements ITrackEvent {
    id: string = '';
    jobId?: string = '';
    jobName?: string = '';
    userId?: string = '';
    actionDate: Date = new Date();
    remoteIP?: string = '';
    remoteResponse?: string = '';
    requestProperties?: string = '';
    source: string = '';
    event: string = '';

    constructor(init?: ITrackEvent) {
        Object.assign(this, init); //this will initialize our object of class JobActions
                                  // with the above values
    }
}