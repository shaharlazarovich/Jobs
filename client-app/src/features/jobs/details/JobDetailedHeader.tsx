import React, { useContext } from 'react'
import { Segment, Item, Button, Header, Image } from 'semantic-ui-react'
import { IJob } from '../../../app/models/job';
import { observer } from 'mobx-react-lite';
import { Link } from 'react-router-dom';
import {format} from 'date-fns'
import { RootStoreContext } from '../../../app/stores/rootStore';

const jobImageStyle = {
  filter: 'brightness(30%)'
};

const jobImageTextStyle = {
  position: 'absolute',
  bottom: '5%',
  left: '5%',
  width: '100%',
  height: 'auto',
  color: 'white'
};

const JobDetailsHeader: React.FC<{job: IJob}> = ({job}) => {
  const rootStore = useContext(RootStoreContext);
  const user = rootStore.userStore.user;
  return (
            <Segment.Group data-test="component-job-detailed-header">
              <Segment basic attached='top' style={{ padding: '0' }}>
                <Image src={`/assets/replicationImages/${job.replication}.jpg`} fluid style={jobImageStyle} />
                <Segment basic style={jobImageTextStyle}>
                  <Item.Group>
                    <Item>
                      <Item.Content>
                        <Header
                          size='huge'
                          content={job.jobName}
                          style={{ color: 'white' }}
                        />
                        <p>{format(job.lastRun, 'eeee do MM')}</p>
                        <p>
                          Created by <Link to={`/profile/${user!.username}`}><strong>{user!.displayName }</strong></Link>
                        </p>
                      </Item.Content>
                    </Item>
                  </Item.Group>
                </Segment>
              </Segment>
              <Segment clearing attached='bottom'>
                  <Button as={Link} to={`/manage/${job.id}`} color='orange' floated='right'>
                    Edit Job
                  </Button>
              </Segment>
            </Segment.Group>
    );
};

export default observer(JobDetailsHeader);