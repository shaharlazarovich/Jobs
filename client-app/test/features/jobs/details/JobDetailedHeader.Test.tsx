import { ReactWrapper } from "enzyme";
import React from "react";
import { IJob } from '../../../../src/app/models/job'
import { findByTestAttr } from '../../../common/testUtils';
import JobDetailedHeader from "../../../../src/features/jobs/details/JobDetailedHeader";
import { jobMock } from "../../../common/mocks";
import { setupWrapper } from "../../../common/wrapper";

describe('Job Detailed Header UniTest', () => {
    let wrapper: ReactWrapper;
    let testJob: IJob;
    beforeEach(async () => {
        testJob = { ...jobMock };
        wrapper = await setupWrapper(<JobDetailedHeader job={testJob} />);
    });
    test('renders without error', () => {
        const component = findByTestAttr(wrapper, 'component-job-detailed-header');
        expect(component.length).toBe(1);
    });
    test('should match the snapshot', () => {
        const component = findByTestAttr(wrapper, 'component-job-detailed-header');
        expect(component).toMatchSnapshot();
    });
})