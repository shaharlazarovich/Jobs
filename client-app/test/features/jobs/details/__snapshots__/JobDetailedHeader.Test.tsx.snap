// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Detailed Header UniTest should match the snapshot 1`] = `
<SegmentGroup
  data-test="component-job-detailed-header"
>
  <div
    className="ui segments"
    data-test="component-job-detailed-header"
  >
    <Segment
      attached="top"
      basic={true}
      style={
        Object {
          "padding": "0",
        }
      }
    >
      <div
        className="ui basic top attached segment"
        style={
          Object {
            "padding": "0",
          }
        }
      >
        <Image
          as="img"
          fluid={true}
          src="/assets/replicationImages/VMware SRM.jpg"
          style={
            Object {
              "filter": "brightness(30%)",
            }
          }
          ui={true}
        >
          <img
            className="ui fluid image"
            src="/assets/replicationImages/VMware SRM.jpg"
            style={
              Object {
                "filter": "brightness(30%)",
              }
            }
          />
        </Image>
        <Segment
          basic={true}
          style={
            Object {
              "bottom": "5%",
              "color": "white",
              "height": "auto",
              "left": "5%",
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <div
            className="ui basic segment"
            style={
              Object {
                "bottom": "5%",
                "color": "white",
                "height": "auto",
                "left": "5%",
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <ItemGroup>
              <div
                className="ui items"
              >
                <Item>
                  <div
                    className="item"
                  >
                    <ItemContent>
                      <div
                        className="content"
                      >
                        <Header
                          content="test1"
                          size="huge"
                          style={
                            Object {
                              "color": "white",
                            }
                          }
                        >
                          <div
                            className="ui huge header"
                            style={
                              Object {
                                "color": "white",
                              }
                            }
                          >
                            test1
                          </div>
                        </Header>
                        <p>
                          Monday 6th 01
                        </p>
                        <p>
                          Created by 
                          <Link
                            to="/profile/Fake Username"
                          >
                            <LinkAnchor
                              href="/profile/Fake Username"
                              navigate={[Function]}
                            >
                              <a
                                href="/profile/Fake Username"
                                onClick={[Function]}
                              >
                                <strong>
                                  Fake Username
                                </strong>
                              </a>
                            </LinkAnchor>
                          </Link>
                        </p>
                      </div>
                    </ItemContent>
                  </div>
                </Item>
              </div>
            </ItemGroup>
          </div>
        </Segment>
      </div>
    </Segment>
    <Segment
      attached="bottom"
      clearing={true}
    >
      <div
        className="ui clearing bottom attached segment"
      >
        <Button
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "Link",
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          color="orange"
          floated="right"
          to="/manage/4"
        >
          <Ref
            innerRef={
              Object {
                "current": <a
                  class="ui orange right floated button"
                  href="/manage/4"
                  role="button"
                >
                  Edit Job
                </a>,
              }
            }
          >
            <RefForward
              innerRef={
                Object {
                  "current": <a
                    class="ui orange right floated button"
                    href="/manage/4"
                    role="button"
                  >
                    Edit Job
                  </a>,
                }
              }
            >
              <Link
                className="ui orange right floated button"
                onClick={[Function]}
                role="button"
                to="/manage/4"
              >
                <LinkAnchor
                  className="ui orange right floated button"
                  href="/manage/4"
                  navigate={[Function]}
                  onClick={[Function]}
                  role="button"
                >
                  <a
                    className="ui orange right floated button"
                    href="/manage/4"
                    onClick={[Function]}
                    role="button"
                  >
                    Edit Job
                  </a>
                </LinkAnchor>
              </Link>
            </RefForward>
          </Ref>
        </Button>
      </div>
    </Segment>
  </div>
</SegmentGroup>
`;
